### PR TITLE
chore(deps): Bumps sample app to RN 0.76.3

### PR DESCRIPTION
**Pull request by antonis**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

Bumps sample app to RN 0.76.3

Refs:
- https://reactnative.dev/blog/2024/10/23/the-new-architecture-is-here
- https://react-native-community.github.io/upgrade-helper/?from=0.75.4&to=0.76.3
- https://reactnative.dev/docs/next/the-new-architecture/pure-cxx-modules

## :bulb: Motivation and Context

[RN 0.76.0 release](https://reactnative.dev/blog/2024/10/23/the-new-architecture-is-here)

Fix https://github.com/getsentry/sentry-react-native/issues/4188

## :green_heart: How did you test it?
Manual Testing with the sample apps, CI
Verify that [the Android app does not crash with profiling enabled](https://github.com/getsentry/sentry-react-native/issues/4188)

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

#skip-changelog

### CODE REVIEW

File: samples/react-native/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "postinstall": "patch-package",
     "android": "react-native run-android",
     "ios": "react-native run-ios",
-    "start": "react-native start --experimental-debugger",
```
Comment (antonis) [inline samples/react-native/package.json:9]: This option has been removed with https://github.com/facebook/react-native/pull/46231

File: samples/react-native/package.json [hunk 2]
```diff
@@ -28,24 +28,25 @@
...
-    "@babel/preset-env": "^7.22.5",
-    "@babel/runtime": "^7.22.5",
-    "@react-native/babel-preset": "0.75.4",
-    "@react-native/eslint-config": "0.75.4",
-    "@react-native/metro-config": "0.75.4",
-    "@react-native/typescript-config": "0.75.4",
+    "@babel/core": "^7.25.2",
+    "@babel/preset-env": "^7.25.3",
+    "@babel/runtime": "^7.25.0",
+    "@react-native-community/cli": "latest",
```
Comment (antonis) [inline samples/react-native/package.json:45]: [This is needed as a dependency now](https://reactnative.dev/blog/2024/10/23/release-0.76-new-architecture#removed-the-dependency-on-the-react-native-communitycli).

File: samples/react-native/package.json [hunk 3]
```diff
@@ -28,24 +28,25 @@
...
-    "react-native": "0.75.4",
-    "react-native-gesture-handler": "^2.18.1",
-    "react-native-reanimated": "3.15.0",
-    "react-native-safe-area-context": "4.10.5",
-    "react-native-screens": "3.34.0",
-    "react-native-svg": "^15.6.0",
+    "react-native": "0.76.0",
+    "react-native-gesture-handler": "^2.20.2",
+    "react-native-reanimated": "3.16.1",
+    "react-native-safe-area-context": "github:th3rdwave/react-native-safe-area-context#main",
```
Comment (antonis) [inline samples/react-native/package.json:34]: The following PR which has not been released yet is needed https://github.com/th3rdwave/react-native-safe-area-context/pull/537

File: samples/react-native/package.json [hunk 4]
```diff
@@ -28,24 +28,25 @@
...
-    "react-native": "0.75.4",
-    "react-native-gesture-handler": "^2.18.1",
-    "react-native-reanimated": "3.15.0",
-    "react-native-safe-area-context": "4.10.5",
-    "react-native-screens": "3.34.0",
-    "react-native-svg": "^15.6.0",
+    "react-native": "0.76.0",
+    "react-native-gesture-handler": "^2.20.2",
+    "react-native-reanimated": "3.16.1",
+    "react-native-safe-area-context": "https://github.com/th3rdwave/react-native-safe-area-context.git#91c426216c18b10558319f8f02383392100b6740",
```
Comment (antonis) [inline samples/react-native/package.json:34]: [This commit](https://github.com/th3rdwave/react-native-safe-area-context/commit/91c426216c18b10558319f8f02383392100b6740) which has not been released yet is needed for 0.76.0

Comment (antonis) [reply samples/react-native/package.json]: Included in `react-native-safe-area-context` `4.12.0`

File: samples/react-native/package.json (hunk 2 above)
Comment (krystofwoldrich) [reply samples/react-native/package.json:45]: The RN template specifies CLI 15, why we use the latest tag?

Comment (antonis) [reply samples/react-native/package.json]: Good catch @krystofwoldrich 🙇 
I've set it with 45e2fafa0c2b0d1cba273f7f40c49746dfbd6f17

**Comment by antonis** [comment]
> (NOBRIDGE) ERROR ReferenceError: An error was thrown when attempting to render log messages via LogBox.
Property 'setImmediate' doesn't exist
> It this still an issue?

@krystofwoldrich yes. The iOS sample compiles successfully and the error occurs on load preventing the app from loading. I guess this is some dependency issue but I haven't figured out a solution yet. Any suggestions are welcome 🙇

File: samples/react-native/src/App.tsx [hunk 5]
```diff
@@ -37,6 +37,10 @@ import HeavyNavigationScreen from './Screens/HeavyNavigationScreen';
 import WebviewScreen from './Screens/WebviewScreen';
 import { isTurboModuleEnabled } from '@sentry/react-native/dist/js/utils/environment';
 
+if (typeof setImmediate === 'undefined') {
+  require('setimmediate');
+}
```
Comment (krystofwoldrich) [inline samples/react-native/src/App.tsx:42]: Why is this required?

**Comment by krystofwoldrich** [comment]
Over all the PR good, thank you for keeping the sample up to date, I just have two questions I want to resolve before approving it.

File: samples/react-native/src/App.tsx (hunk 5 above)
Comment (antonis) [reply samples/react-native/src/App.tsx:42]: This is a workaround for the [set `setImmediate` error](https://github.com/getsentry/sentry-react-native/pull/4207#issuecomment-2462139171) on iOS mentioned before. I haven't figured out the root of the issue.

File: samples/react-native/tm/NativePlatformSampleModule.ts [hunk 6]
```diff
@@ -1,8 +1,7 @@
-import type { TurboModule } from 'react-native';
-import { TurboModuleRegistry } from 'react-native';
+import {TurboModule, TurboModuleRegistry} from 'react-native';
 
 export interface Spec extends TurboModule {
   crashOrString(): string;
 }
 
-export default TurboModuleRegistry.get<Spec>('NativePlatformSampleModule');
+export default TurboModuleRegistry.getEnforcing<Spec>('NativePlatformSampleModule');
```
[the remark this answers is not in the document]
Comment (antonis) [reply samples/react-native/tm/NativePlatformSampleModule.ts]: Thank you for the context @krystofwoldrich
Reverted with db06022bba59676e820d221cb585f620308206e8

File: samples/react-native/src/App.tsx (hunk 5 above)
Comment (krystofwoldrich) [reply samples/react-native/src/App.tsx:42]: Aaa, my bad, yes, you described in the comments above.

**Comment by krystofwoldrich** [comment]
The missing `setImmediate` might be related to the Promises patch the SDK automatically applies. My guess is the error will go away when the Promise patch is disabled.

**Comment by antonis** [comment]
> The missing setImmediate might be related to the Promises patch the SDK automatically applies. My guess is the error will go away when the Promise patch is disabled.

Thank you for pointing me to this @krystofwoldrich 🙇 

I've tried with [the `patchGlobalPromise` hardcoded to `false`](https://github.com/getsentry/sentry-react-native/blob/antonis/chore-samples-rn0.76.0/samples/react-native/src/App.tsx#L100) and the error persists 🤔

**Review by krystofwoldrich — APPROVED** [review]
LGTM 🚀
